phpstan.neon.dist

```diff
@@ -27,6 +27,7 @@ parameters:
 		- '#Call to an undefined method CodeIgniter\\HTTP\\Request::(getPath|getSegments|getMethod|setLocale|getPost)\(\)#'
 		- '#Call to an undefined method CodeIgniter\\Router\\RouteCollectionInterface::(getDefaultNamespace|isFiltered|getFilterForRoute|getRoutesOptions)\(\)#'
 		- '#Call to an undefined method CodeIgniter\\Validation\\ValidationInterface::loadRuleGroup\(\)#'
+		- '#Call to an undefined static method CodeIgniter\\Config\\Factories::[a-z]+\(\)#'
```

Comment:
I'm curious what the actual error message is. Maybe we can add `Factories` as a [universal object crate class](https://phpstan.org/config-reference#universal-object-crates) or adding the `@property type $name` to the expected properties of this one.

Reply:
Ooo this is great! I didn't know about the crate objects. Now I have to update like 15 repos 😢

Reply:
Actually, crate classes are only concerned with properties, not methods. Same with the `@property` tag. I poked around the PHPStan repo and there was some recent movement on this issue (https://github.com/phpstan/phpstan/issues/3641) but the solutions are either to `mixin`  another class (not applicable) or declare the `@method`, which might work for our explicit calls but will not cover the generic uses like `Factories::foo('bar')` where you don't know "foo" ahead of time.

Here are the actual errors without the ignore:
```
 ------ ----------------------------------------- 
  Line   system/Common.php                        
 ------ ----------------------------------------- 
  256    Call to an undefined static method       
         CodeIgniter\Config\Factories::config().  
  872    Call to an undefined static method       
         CodeIgniter\Config\Factories::models().  
 ------ ----------------------------------------- 

 ------ ----------------------------------------- 
  Line   system/Config/Config.php                 
 ------ ----------------------------------------- 
  60     Call to an undefined static method       
         CodeIgniter\Config\Factories::config().  
 ------ ----------------------------------------- 

 ------ ----------------------------------------- 
  Line   system/Database/ModelFactory.php         
 ------ ----------------------------------------- 
  24     Call to an undefined static method       
         CodeIgniter\Config\Factories::models().  
 ------ ----------------------------------------- 
```

Reply:
Confirmed, adding the following to the class docblock does prevent the errors:
```
 * @method static \CodeIgniter\Model models(...$arguments)
 * @method static \Config\BaseConfig config(...$arguments)
```
We'll go with that for now but it won't solve the issue for projects downstream.